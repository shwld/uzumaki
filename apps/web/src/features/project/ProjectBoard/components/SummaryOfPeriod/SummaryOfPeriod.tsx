import { FC } from 'react';
import { HStack, ListItemProps, ListItem, Text } from '@chakra-ui/react';
import { CalendarIcon } from '@chakra-ui/icons';
import dayjs from 'dayjs';

export const SummaryOfPeriod: FC<
  {
    points: number;
    startDate: Date;
    iterationLength: number;
  } & Omit<ListItemProps, 'children'>
> = ({ points, startDate, iterationLength, ...props }) => {
  const endDate = dayjs(startDate).add(iterationLength, 'weeks');
  return (
    <ListItem
      borderBottom="1px"
      backgroundColor="blue.600"
      py={0}
      px={2}
      {...props}
    >
      <HStack justify="space-between" align="center">
        <HStack align="center">
          <CalendarIcon color="gray.100" />
          <Text color="gray.100" fontSize="sm">
            {points} points
          </Text>
        </HStack>
        <HStack align="center">
          <Text color="gray.100" fontSize="sm">
            {dayjs(startDate).format('DD MMM')} -{' '}
            {dayjs(endDate).format('DD MMM')}
          </Text>
        </HStack>
      </HStack>
    </ListItem>
  );
};
