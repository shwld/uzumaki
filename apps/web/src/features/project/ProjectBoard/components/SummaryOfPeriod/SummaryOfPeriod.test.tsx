import { SummaryOfPeriod } from './SummaryOfPeriod';
import { render } from '@testing-library/react';
import { List } from '@chakra-ui/react';

describe('SummaryOfPeriod', () => {
  const renderComponent = () => {
    const renderResult = render(
      <List>
        <SummaryOfPeriod
          points={10}
          startDate={new Date('2022-2-2')}
          iterationLength={2}
        />
      </List>
    );
    return renderResult;
  };
  test('Snapshot', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('success', () => {
    const { getByText } = renderComponent();
    expect(getByText('10 points')).toBeTruthy();
  });
});
