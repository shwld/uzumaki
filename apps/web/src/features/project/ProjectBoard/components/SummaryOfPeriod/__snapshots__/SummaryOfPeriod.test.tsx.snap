// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryOfPeriod Snapshot 1`] = `
<DocumentFragment>
  <ul
    class="css-155za0w"
    role="list"
  >
    <li
      class="css-lybpxv"
    >
      <div
        class="chakra-stack css-jnbq08"
      >
        <svg
          class="chakra-icon css-17hpw6g"
          focusable="false"
          viewBox="0 0 14 14"
        >
          <path
            d="M10.8889,5.5 L3.11111,5.5 L3.11111,7.05556 L10.8889,7.05556 L10.8889,5.5 Z M12.4444,1.05556 L11.6667,1.05556 L11.6667,0 L10.1111,0 L10.1111,1.05556 L3.88889,1.05556 L3.88889,0 L2.33333,0 L2.33333,1.05556 L1.55556,1.05556 C0.692222,1.05556 0.00777777,1.75556 0.00777777,2.61111 L0,12.5 C0,13.3556 0.692222,14 1.55556,14 L12.4444,14 C13.3,14 14,13.3556 14,12.5 L14,2.61111 C14,1.75556 13.3,1.05556 12.4444,1.05556 Z M12.4444,12.5 L1.55556,12.5 L1.55556,3.94444 L12.4444,3.94444 L12.4444,12.5 Z M8.55556,8.61111 L3.11111,8.61111 L3.11111,10.1667 L8.55556,10.1667 L8.55556,8.61111 Z"
            fill="currentColor"
          />
        </svg>
        <p
          class="chakra-text css-1aq3av"
        >
          10 points
        </p>
        <div
          class="chakra-stack css-84zodg"
        />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
